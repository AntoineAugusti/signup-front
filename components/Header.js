import Link from 'next/link';
import React from 'react';

const Header = () => (
  <header>
    <nav className="nav">
      <div className="nav__container">
        <a className="nav__link" href="index.html">
          <img className="nav__logo" src="static/images/logo_site.svg" alt="Accueil de particulier.api.data.gouv.fr" />
        </a>
        <ul className="nav__links">
          <li><Link href="/"><a>Home</a></Link></li>
          <li><Link href="/about"><a>About</a></Link></li>
        </ul>
      </div>
    </nav>
  </header>
);

export default Header;
