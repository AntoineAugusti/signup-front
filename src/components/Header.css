.back-button {
  border: 0;
  display: inline-block;
  background-color: inherit;
  font-size: 1.5em;
  color: var(--black);
}

.back-button:hover {
  background-color: var(--lighter-grey);
}

.back-button:focus {
  outline: 0;
}
