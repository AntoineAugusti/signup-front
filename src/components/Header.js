import React from 'react';
import PropTypes from 'prop-types';
import { withUser } from './UserContext';

const Header = ({ user, logout }) => {
  const displayContactLink =
    window.location.pathname === '/' ||
    window.location.pathname === '/stats' ||
    window.location.pathname.startsWith('/public');

  return (
    <header className="navbar">
      <div className="navbar__container">
        <a className="navbar__home" href="https://api.gouv.fr/?filter=signup">
          <img
            className="navbar__logo"
            src="/images/logo_api.svg"
            alt="Accueil de signup.api.gouv.fr"
          />
        </a>

        <nav>
          <ul className="nav__links">
            <li className="nav__item">
              <a href="https://api.gouv.fr/apropos">À propos</a>
            </li>
            {displayContactLink && (
              <li className="nav__item">
                <a href="mailto:contact@particulier.api.gouv.fr?subject=Contact%20via%20signup.api.gouv.fr">
                  Nous contacter
                </a>
              </li>
            )}
            {user && (
              <li className="nav__item">
                <div className="dropdown">
                  {user.email}
                  <div className="dropdown-content" style={{ top: '1.8em' }}>
                    <a onClick={logout} href="#logout">
                      Se déconnecter
                    </a>
                  </div>
                </div>
              </li>
            )}
          </ul>
        </nav>
      </div>
    </header>
  );
};

Header.propTypes = {
  user: PropTypes.object,
  logout: PropTypes.func.isRequired,
};

Header.defaultProps = {
  user: null,
};

export default withUser(Header);
