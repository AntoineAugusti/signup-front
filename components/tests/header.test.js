import React from 'react'
import ReactTestRenderer from 'react-test-renderer'
import Header from '../header'
import attachUser from '../../components/hoc/attach-user'

require('../../lib/tests/local-storage') // eslint-disable-line import/no-unassigned-import

class Component extends React.Component {
  render() {
    const {children} = this.props

    return <div>{children}</div>
  }
}

const AttachedComponent = attachUser(Component)

describe('components | Header', () => {
  describe('render', () => {
    const renderer = ReactTestRenderer.create(<AttachedComponent><Header user={function () {}}/></AttachedComponent>)
    it('should be defined', () => {
      expect(renderer).toBeDefined()
    })
    it('should match snapshot', () => {
      expect(renderer).toMatchSnapshot()
    })
  })
})
