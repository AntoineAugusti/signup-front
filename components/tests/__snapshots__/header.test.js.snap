// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components | Header render should match snapshot 1`] = `
<div>
  <header>
    <nav
      className="nav"
    >
      <div
        className="nav__container"
      >
        <a
          className="nav__link"
          href="/"
        >
          <img
            alt="Accueil de particulier.api.data.gouv.fr"
            className="nav__logo"
            src="/static/images/logo-api-particulier.png"
          />
        </a>
        <ul
          className="nav__links"
        >
          <li>
            <a
              href="/#france-connect"
              onClick={[Function]}
            >
              Donn√©es disponibles
            </a>
          </li>
          <li>
            <a
              href="http://localhost:3022/oauth/authorize?client_id=82546188522214c3577d35c283ce8842786649b35a026a9d44908037a597f29b&redirect_uri=http%3A%2F%2Flocalhost%3A3000&scope=&response_type=token&state="
            >
              Se connecter
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </header>
</div>
`;
